spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -1925,6 +1925,11 @@ def fix_indent(self, forward=True, comment_or_string=False):
             cursor.movePosition(QTextCursor.PreviousBlock)
             prevtext = to_text_string(cursor.block().text()).rstrip()
 
+            # Remove inline comment
+            inline_comment = prevtext.find('#')

```

Comment:
`prevtext = ""` and the indentation will be calculated to `0` as expected.

I was testing and It fails in expressions with `#` that isn't a comment, and the crop cause to drop a bracket, although it is pretty rare (e. g. a list assignment to a dict element that contains a "#"):

`a_dict["#"] = (`

```
a_dict["#"] = (
       |
```